ssl/ssl_ciph.c

```diff
@@ -995,14 +995,8 @@ static int ssl_cipher_process_rulestr(const char *rule_str,
             ch = *l;
             buf = l;
             buflen = 0;
-#ifndef CHARSET_EBCDIC
-            while (((ch >= 'A') && (ch <= 'Z')) ||
-                   ((ch >= '0') && (ch <= '9')) ||
-                   ((ch >= 'a') && (ch <= 'z')) ||
-                   (ch == '-') || (ch == '.') || (ch == '='))
-#else
-            while (isalnum(ch) || (ch == '-') || (ch == '.') || (ch == '='))
-#endif
+            while (!(ch & ~127) && (isalnum(ch) || ch == '-' || ch == '.'
+                                    || ch == '='))
```

Comment:
But there is no inconsistency *here*. Caller and the rest of the function in question operate in native encoding.